auto_dev/data/templates/protocols/test_dialogues.jinja

```diff
@@ -0,0 +1,73 @@
+{{ header }}
+
+"""Test dialogues module for the {{ snake_name }} protocol."""
+
+from unittest.mock import MagicMock
+
+from pydantic import BaseModel, conint, confloat
+from hypothesis import given
+from hypothesis import strategies as st
+from aea.configurations.data_types import PublicId
+
+from packages.{{ author }}.protocols.{{ snake_name }}.dialogues import (
+    {{ camel_name }}Dialogue,
+    {{ camel_name }}Dialogues,
+)
+from packages.{{ author }}.protocols.{{ snake_name }}.message import {{ camel_name }}Message
+from packages.{{ author }}.protocols.{{ snake_name }}.tests.primitive_strategies import (
+    Int32,
+    Double,
+)
+from packages.{{ author }}.protocols.{{ snake_name }}.custom_types import (
+    {%- for custom_type in custom_types %}
+    {{ custom_type }},
+    {%- endfor %}
+)
+
+
+def shallow_dump(model: BaseModel) -> dict:
+    """Shallow dump pydantic model."""
+
+    return {name: getattr(model, name) for name in model.__fields__}
```

Comment:
this allows to generate pydantic models (as we do for to .proto messages) for the performatives (which are in fact also .proto messages, but they are a custom addition by framework and handled in a framework specific manner).  Defining them as pydantic models, tho, massively simplifies their hypothesis strategy generation (via `st.from_type()`), however we need to `shallow_dump` to pass the `kwargs` to `dialogues.create`.